xarray/tests/test_netcdftimeindex.py

```diff
@@ -0,0 +1,363 @@
+import pytest
+
+import pandas as pd
+import xarray as xr
+
+from xarray.conventions.netcdftimeindex import (
+    parse_iso8601, NetCDFTimeIndex, assert_all_same_netcdftime_datetimes,
+    _parsed_string_to_bounds, _parse_iso8601_with_reso)
+from xarray.tests import assert_array_equal, assert_identical
+
+# Putting this at the module level for now, though technically we
+# don't need netCDF4 to test the string parser.
+pytest.importorskip('netCDF4')
+
+
+def date_dict(year=None, month=None, day=None,
+              hour=None, minute=None, second=None):
+    return dict(year=year, month=month, day=day, hour=hour,
+                minute=minute, second=second)
+
+ISO8601_STRING_TESTS = [
```

Comment:
Is this along the lines of what you were looking for here?  I couldn't find a name for the second argument to ```pytest.mark.parametrize```, so I wasn't sure how to combine these two list arguments into a dict (but maybe I misunderstood what you were asking for).